kernel/examples/kernel-examples/src/main/java/io/delta/kernel/examples/CreateTableAndInsertData.java

```diff
@@ -0,0 +1,423 @@
+    }
+
+    public TransactionCommitResult createTableWithSampleData(String tablePath) throws IOException {
+        // Create a `Table` object with the given destination table path
+        Table table = Table.forPath(engine, tablePath);
+
+        // Create a transaction builder to build the transaction
+        TransactionBuilder txnBuilder =
+                table.createTransactionBuilder(
+                        engine,
+                        "Examples", /* engineInfo */
+                        Operation.CREATE_TABLE);
+
+        // Set the schema of the new table on the transaction builder
+        txnBuilder = txnBuilder.withSchema(engine, exampleTableSchema);
+
+        // Build the transaction
+        Transaction txn = txnBuilder.build(engine);
+
+        // Get the transaction state
+        Row txnState = txn.getTransactionState(engine);
+
+        // Generate the sample data for the table that confirms to the table schema
+        FilteredColumnarBatch batch1 = generateUnpartitionedDataBatch(5 /* offset */);
+        FilteredColumnarBatch batch2 = generateUnpartitionedDataBatch(10 /* offset */);
+        FilteredColumnarBatch batch3 = generateUnpartitionedDataBatch(25 /* offset */);
+        CloseableIterator<FilteredColumnarBatch> data =
+                toCloseableIterator(Arrays.asList(batch1, batch2, batch3).iterator());
+
+        // First transform the logical data to physical data that needs to be written to the Parquet
+        // files
+        CloseableIterator<FilteredColumnarBatch> physicalData =
+                Transaction.transformLogicalData(
+                        engine,
+                        txnState,
+                        data,
+                        // partition values - as this table is unpartitioned, it should be empty
+                        Collections.emptyMap());
+
+        // Get the write context
+        DataWriteContext writeContext = Transaction.getWriteContext(
+                engine,
+                txnState,
+                // partition values - as this table is unpartitioned, it should be empty
+                Collections.emptyMap());
+
+
+        // Now write the physical data to Parquet files
+        CloseableIterator<DataFileStatus> dataFiles = engine.getParquetHandler()
+                .writeParquetFiles(
+                        writeContext.getTargetDirectory(),
+                        physicalData,
+                        writeContext.getStatisticsColumns());
+
+
+        // Now convert the data file status to data actions that needs to be written to the Delta
+        // table log
+        CloseableIterator<Row> dataActions =
+                Transaction.generateAppendActions(engine, txnState, dataFiles, writeContext);
+
+
+        // Create a iterable out of the data actions. If the contents are too big to fit in memory,
+        // the connector may choose to write the data actions to a temporary file and return an
```

Comment:
just curious: how will they know if it is too big to fit into memory? you don't know how big the iterator is

Reply:
Good qn. Start with a buffer and consume the entries from the tasks. If the buffer is filled, start spilling to disks.